mso/resource_mso_schema_site_vrf_route_leak.go

```diff
@@ -0,0 +1,252 @@
+								d.Set("type", "leak_all")
+							} else {
+								d.Set("type", "subnet_ip")
+							}
+							d.Set("subnet_ips", subnetIps)
+							return nil
+						}
+					}
+				}
+			}
+		}
+	}
+	d.SetId("")
+	return fmt.Errorf("Unable to find route leak information for the Site Vrf %s", vrfName)
+}
+
+func resourceMSOSchemaSiteVrfRouteLeakImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	msoClient := m.(*client.Client)
+	splitImport := strings.Split(d.Id(), "/")
+	schemaId := splitImport[0]
+	schemaCont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+	if err != nil {
+		return nil, err
+	}
+	err = setRouteLeakFromSchema(d, schemaCont, schemaId, splitImport[2], splitImport[4], splitImport[6], getVrfRef(splitImport[8], splitImport[9], splitImport[10]))
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceMSOSchemaSiteVrfRouteLeakCreate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Create", d.Id())
+	msoClient := m.(*client.Client)
+	siteId := d.Get("site_id").(string)
+	prefixSubnets, includeAllSubnets := getSubnetDetails(d)
+	path := fmt.Sprintf("/sites/%s-%s/vrfs/%s/routeLeak/-", d.Get("site_id").(string), d.Get("template_name").(string), d.Get("vrf_name").(string))
+	vrfRouteLeakStruct := models.NewSchemaSiteVrfRouteLeak(
+		"add", path, d.Get("tenant_name").(string), getTargetVrfRef(d), includeAllSubnets, prefixSubnets, []string{siteId},
+	)
+	_, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", d.Get("schema_id").(string)), vrfRouteLeakStruct)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Create finished successfully", d.Id())
+	return resourceMSOSchemaSiteVrfRouteLeakRead(d, m)
+}
+
+func resourceMSOSchemaSiteVrfRouteLeakUpdate(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Update", d.Id())
+	msoClient := m.(*client.Client)
+	prefixSubnets, includeAllSubnets := getSubnetDetails(d)
+	vrfRegionStruct := models.NewSchemaSiteVrfRouteLeak(
+		"replace", d.Id(), d.Get("tenant_name").(string), getTargetVrfRef(d), includeAllSubnets, prefixSubnets, []string{d.Get("site_id").(string)},
+	)
+	_, err := msoClient.PatchbyID(fmt.Sprintf("api/v1/schemas/%s", d.Get("schema_id").(string)), vrfRegionStruct)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Update finished successfully", d.Id())
+	return resourceMSOSchemaSiteVrfRouteLeakRead(d, m)
+}
+
+func resourceMSOSchemaSiteVrfRouteLeakRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+	msoClient := m.(*client.Client)
+	schemaId := d.Get("schema_id").(string)
+	schemaCont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", schemaId))
+	if err != nil {
+		return err
```

Comment:
Please use errorForObjectNotFound to handle the 404 error.
```suggestion
		return errorForObjectNotFound(err, dn, con, d)
```